packages/admin-next/dashboard/src/routes/products/product-edit-variant/components/product-edit-variant-form/product-edit-variant-form.tsx

```diff
@@ -0,0 +1,454 @@
+import { CountrySelect } from "../../../../../components/common/country-select"
+import { Divider } from "../../../../../components/common/divider"
+import { Form } from "../../../../../components/common/form"
+import { RouteDrawer } from "../../../../../components/route-modal"
+import { castNumber } from "../../../../../lib/cast-number"
+import { optionalInt } from "../../../../../lib/validation"
+
+type ProductEditVariantFormProps = {
+  product: Product
+  variant: ProductVariant
+  isStockAndInventoryEnabled?: boolean
+}
+
+const ProductEditVariantSchema = z.object({
+  title: z.string().min(1),
+  material: z.string().optional(),
+  sku: z.string().optional(),
+  ean: z.string().optional(),
+  upc: z.string().optional(),
+  barcode: z.string().optional(),
+  inventory_quantity: optionalInt,
+  manage_inventory: z.boolean(),
+  allow_backorder: z.boolean(),
+  weight: optionalInt,
+  height: optionalInt,
+  width: optionalInt,
+  length: optionalInt,
+  mid_code: z.string().optional(),
+  hs_code: z.string().optional(),
+  origin_country: z.string().optional(),
+})
+
+export const ProductEditVariantForm = ({
+  product,
+  variant,
+  isStockAndInventoryEnabled = false,
+}: ProductEditVariantFormProps) => {
+  const { t } = useTranslation()
+
+  const form = useForm<z.infer<typeof ProductEditVariantSchema>>({
+    defaultValues: {
+      title: variant.title,
+      material: variant.material || "",
+      sku: variant.sku || "",
+      ean: variant.ean || "",
+      upc: variant.upc || "",
+      barcode: variant.barcode || "",
+      inventory_quantity: variant.inventory_quantity || "",
+      manage_inventory: variant.manage_inventory,
+      allow_backorder: variant.allow_backorder,
+      weight: variant.weight || "",
+      height: variant.height || "",
+      width: variant.width || "",
+      length: variant.length || "",
+      mid_code: variant.mid_code || "",
+      hs_code: variant.hs_code || "",
+      origin_country: variant.origin_country || "",
+    },
+    resolver: zodResolver(ProductEditVariantSchema),
+  })
+
+  const { mutateAsync, isLoading } = useAdminUpdateVariant(product.id)
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    const parseNumber = (value?: string | number) => {
+      if (typeof value === "undefined" || value === "") {
+        return undefined
+      }
+
+      if (typeof value === "string") {
+        return castNumber(value)
+      }
+
+      return value
+    }
+
+    const {
+      weight,
+      height,
+      width,
+      length,
+      inventory_quantity,
+      allow_backorder,
+      manage_inventory,
+      sku,
+      ean,
+      upc,
+      barcode,
+      ...rest
+    } = data
+
+    /**
+     * If stock and inventory is not enabled, we need to send the inventory and
+     * stock related fields to the API. If it is enabled, it should be handled
+     * in the separate stock and inventory form.
+     */
+    const conditionalPayload = !isStockAndInventoryEnabled
```

Comment:
q: Didn't we discuss that we assume that inventory/stock location modules are enabled for this version? Asking because I have similar situations with fulfilment and I'm not sure about the approach.

Reply:
I believe this was for 2.0. At least I think it should be. I don't think we can enforce this, as that would be a significant upgrade for users to go through to get a new version of admin. And we will communicate that the admin redesign is fully compatible with setups that run on 1.*, which includes simple inventory management.

Reply:
I understood it as it not being needed for the alpha/beta version. But we should have support for both cases for the final release, and I added it here as the check is rather simple.